pkg/apiserver/domain/service/workflow.go

```diff
@@ -789,19 +789,36 @@ func (w *workflowServiceImpl) TerminateRecord(ctx context.Context, appModel *mod
 }
 
 // ResumeWorkflow resume workflow
-func ResumeWorkflow(ctx context.Context, kubecli client.Client, app *v1beta1.Application) error {
+func ResumeWorkflow(ctx context.Context, kubecli client.Client, app *v1beta1.Application, stepName string) error {
 	app.Status.Workflow.Suspend = false
 	steps := app.Status.Workflow.Steps
+	found := stepName == ""
+
 	for i, step := range steps {
 		if step.Type == wfTypes.WorkflowStepTypeSuspend && step.Phase == workflowv1alpha1.WorkflowStepPhaseRunning {
```

Comment:
If we add a provider function to suspend the current step, the step type that needs to resume is not only `suspend`. Maybe we should remove the condition `step.Type == wfTypes.WorkflowStepTypeSuspend` when the step name is not empty. What do you think?

Reply:
Currently, we don't have this `pause/suspend` provider and only `suspend` step can work. I think we can change the condition when we have such capabilities. Besides, we need to change a lot if we add such providers, eg, maybe add some phase like `suspending`.